presets/presetManager.py

```diff
@@ -0,0 +1,223 @@
+import os.path
+import logging
+import json
+
+logger = logging.getLogger(__name__)
+
+
+class PresetManager(object):
+    MAX_DEPTH = 2
+
+    def __init__(self, paths, strict=False):
+        self.presets = {}
+        self.strict = strict
+
+        # normalize paramter type string -> array
+        if isinstance(paths, basestring):
+            paths = [paths]
+
+        self._loadPaths(paths)
+
+    def _loadPaths(self, paths, depth=0):
+        if depth > self.MAX_DEPTH:
+            return
+
+        for i in range(len(paths)):
+            path = paths[i]
```

Comment:
Which side effect are you talking of?
It's not a matter of performance, which would be irrelevant. It's a matter of readability, which is way more important!

If you think that there is a good reason to write code like this, you should at least explain **why** in a comment. Any pythonista will look at this code wondering if it is just badly written, or if there is some specific reason it's written like this, which they are not being able to understand.
I still have not, for example :)

Other similar readability issues are commented separately